src/main/java/com/example/AllPermutationPrime.java

```diff
@@ -0,0 +1,62 @@
+package com.example;
+
+import java.util.ArrayList;
+
+/**
+ * Készíts egy olyan függvényt, aminek
+ * bemenő paramétere n,
+ * és válaszként kilistázza azokat a számokat 1-től n-ig,
+ * amelyeknek az összes körkörös permutációja prím.
+ */
+public class AllPermutationPrime {
+
+    private static boolean isPrime(int n) {        
+        boolean prime = true;
+        int divider = n - 1;
+        while ((divider > 1) && prime) {
+            if (n % divider == 0) {
+                prime = false;
+            }
+            else {
+                divider--;
+            }
+        }
+        return prime;
+    }
+
+    private static int[] generateCircledPermutations(int checkedNumber) {
```

Comment:
You could use `List` as a return type.

Reply:
Easily.
In such cases, might increase memory need? Slower performance? Primitive typed arrays are faster than their wrapper typed Lists?

Reply:
Rule of thumb: 
- Maintainability, readability, flexibility trumps performance most of the time
- optimize for performance when you have to, most of the time you can win much more if you find cases where you don't have to call a function at all (wins 100%) 
- only take into account performance aspects if you can prove there is a difference

Reply:
If I understand correctly:

- use Lists because they are elegant (my opinion) but forget performance drop,
- don't use Lists because whatever you do with them, you will call methods (opposite: array.length or array[index] are not methods),
- aware of recursions (the absolute winners in calling methods), but they are among interview questions.

Some paradoxes. :)

No, I understand you perfectly, code complexity should be low, should try to organize solid tasks into methods, but should keep balance with low amount of methods calles, or something like that.

Ultimately, as I see, it is a decision-competition between the developer of a task, the developer who accepts the work, and the project time available for closing the task. :)

Thx!